lang/java/reef-tang/tang/src/main/java/org/apache/reef/tang/implementation/java/JavaNodeFactory.java

```diff
@@ -142,25 +142,27 @@
   /**
    * XXX: This method assumes that all generic types have exactly one type parameter.
    */
-  public static <T> NamedParameterNode<T> createNamedParameterNode(Node parent,
-                                                                   Class<? extends Name<T>> clazz, Type argClass)
+  public static <T> NamedParameterNode<T> createNamedParameterNode(final Node parent,
+                                                                   final Class<? extends Name<T>> clazz,
+                                                                   final Type argClass)
       throws ClassHierarchyException {
 
-    Class<?> argRawClass = ReflectionUtilities.getRawClass(argClass);
+    Type argClazz = argClass;
 
-    final boolean isSet = argRawClass.equals(Set.class);
-    final boolean isList = argRawClass.equals(List.class);
+    Class<?> argRawClazz = ReflectionUtilities.getRawClass(argClazz);
```

Comment:
I'm not sure if we should, since argClazz is reassigned. I don't know enough about this code to be confident to make that change.